pkg/exporter/buffered.go

```diff
@@ -68,8 +68,21 @@ func NewBufferedIPFIXExporter(ep *ExportingProcess) *BufferedIPFIXExporter {
 }
 
 func (e *BufferedIPFIXExporter) addTemplateRecord(record entities.Record) error {
+	templateID := record.GetTemplateID()
+	// If the templateID already exists, we should send corresponding buffered data records
+	// immediately, as they may not match the new template definition.
+	if m, ok := e.messages[templateID]; ok {
+		if err := m.flush(); err != nil {
```

Comment:
Should we return after flushing? Otherwise I think we will have duplicate records?

Reply:
no, the new template should be sent and will override the old template, both locally in the exporter (`templateMap`) and at the destination collector